src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/ExecutableStatementCountCheck.java

```diff
@@ -95,7 +95,6 @@ public void setMax(int max) {
     @Override
     public void beginTree(DetailAST rootAST) {
         context = new Context(null);
-        contextStack.clear();
```

Comment:
> Let's make a method resetCheckInternalState and such non required cleanup in it, additionally configure pitest to skip such method at all.

Since the methods are public why can we do something like:
`````
DetailAST METHOD_DEF = {create ast};
ExecutableStatementCountCheck instance = new ExecutableStatementCountCheck();
instance.visitToken(METHOD_DEF);
instance.beginTree(null);
// verify contextStack in instance is cleared
````
Since this is default for all AbstractChecks, you might be able to make a test utility out of this to test these weird conditions on all file state fields that must be reset with each new file.